cpp/include/cudf/io/detail/batched_memcpy.hpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/detail/iterator.cuh>
+#include <cudf/utilities/memory_resource.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_buffer.hpp>
+
+#include <cub/device/device_memcpy.cuh>
+#include <cuda/functional>
+#include <thrust/iterator/constant_iterator.h>
+
+namespace CUDF_EXPORT cudf {
+namespace io::detail {
```

Comment:
this does not need to be under io, might be useful for other parts of libcudf.

Reply:
Only put it here to be alongside the `batched_memset.hpp` utility. Though should we combine them in one file?

Reply:
I'm not convinced they should be in the same file, at not lot with current implementations (pretty mismatched :)).
I'd be fine with a simple move out of io for now.

Reply:
Sounds good to me. I should then move both `batched_memxxx` out of IO then.